eng/pipelines/templates/steps/build-test.yml

```diff
@@ -53,6 +53,13 @@ steps:
         ${{ parameters.ToxEnvParallel }}
     env: ${{ parameters.EnvVars }}
 
+  - task: PublishPipelineArtifact@1
+    condition: failed()
```

Comment:
Don't these need an artifactName as well?

Reply:
No, it's uploading a folder and Azure Pipelines will create the new folder and upload all the contents from the folder given.

Reply:
What name does it end up giving the artifact?

Reply:
It will be named after the job name, Python39 for example.